### DISCUSSION TITLE
Drayton Wiser WV704R0A0902

### DISCUSSION THREAD

**Discussion by paul-jewell** [Koenkk/zigbee2mqtt#6864] [Devices]
I have installed zigbee2mqtt, and in order to find my way around installing and managing devices I have installed one TRV with the above part number. I have a couple of issues which could well be a result of my low level understanding!
I connected the device using the instructions in the documentation - all good.
These are the issues I now have:
1. The Device has been recognised correctly by Home Assistant, and a Thermostat card added to the default lovelace dashboard. On this card, I can set the target temperature, and see the current temperature. There are also three buttons for setting the mode to Auto, Heat and Off. However, when I try to set the mode, I see an error in the zigbee2mqtt logs:
 "No converter available for 'system_mode' (heat)"
According to the documentation, this should work?
2. Maybe related to point 1, and not being able to change the mode. I set the target temperature to 17°C (for example), and at some point later the target is set to 20°C. However, it isn't clear what is doing this. I don't see any MQTT messages, and I haven't set up any automation which could be making this change. I suspect the device is in Auto mode, and this is a feature built into that mode?

I would like to resolve (1), so if anyone can give me some pointers, I would be very grateful!

One final question - in the documentation under device specific settings, there is a section saying set "legacy: false". In the configuration page, I don't see a key called legacy, but one called legacy_api. Is this the correct key?

**Comment by sjorge** [comment]
I don’t think system_mode works for these devices, I have the Eberly branded one but both are white label SE. Mine also return unsupported attribute when setting a system_mode.

**Reply by paul-jewell** [reply]
Thanks for the comment @sjorge. Do you have any issues with the temperature setpoint being changed autonomously?

**Reply by sjorge** [reply]
It seems to reset after a certain interval yes.

I have a bunch of TRVs and I sadly have none that I can recommend, as I have issues with all. Including this one:

**Reply by paul-jewell** [reply]
It is quite frustrating! All I want is a trv i can switch on and off, or better still set the valve open point between open and closed. I have separate room thermostats, and I would be quite happy to write the control logic to manage the temperature. From comments I had seen, this particular valve seemed to be the best solution available. I could even make something workable if I didn't have the problem of the device changing its own set point!

I guess I will have to keep looking...

**Reply by paul-jewell** [reply]
I've sent mine back to Amazon - I did try a second on to see if there was something causing the first to reset the target temperature, but the behaviour was the same. I guess I am going to have to look elsewhere.

**Comment by adamlc** [comment]
I've just bought one of these to try out. It seems the problem still persists with these things!

It worked fine for a couple of days and would let me set the temperature, but since then it just either ignores any changes or just resets.
